src/json.ts

```diff
@@ -0,0 +1,137 @@
+        return result;
+    }
+
+    function peek(offset: number = 0): number | undefined {
+        if ((index + offset) < length) {
+            return text.charCodeAt(index + offset);
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    function scanString() {
+        while (true) {
+            if (index >= length) { // string ended unexpectedly
+                break;
+            }
+
+            let code = next();
+
+            if (code === CharCode.doubleQuote) {
+                // End of string. We're done
+                break;
+            }
+
+            if (code === CharCode.backSlash) {
+                // Skip escaped character. We don't care about verifying the escape sequence.
+                // We just don't want to accidentally scan an escaped double-quote as the end of the string.
+                index++;
+            }
+
+            if (isLineBreak(code)) {
+                // string ended unexpectedly
+                break;
+            }
+        }
+    }
+
+    while (true) {
+        let code = next();
+
+        switch (code) {
+            // strings
+            case CharCode.doubleQuote:
+                scanString();
+                break;
+
+            // comments
+            case CharCode.slash:
+                // Single-line comment
+                if (peek() === CharCode.slash) {
+                    // Be careful not to include the first slash in the text part.
+                    parts.push(text.substring(partStart, index - 1));
+
+                    // Start after the second slash and scan until a line-break character is encountered.
+                    index++;
+                    while (index < length) {
+                        if (isLineBreak(peek())) {
+                            break;
+                        }
+
+                        index++;
+                    }
+
+                    partStart = index;
+                }
+
+                // Multi-line comment
+                if (peek() === CharCode.asterisk) {
+                    // Be careful not to include the first slash in the text part.
+                    parts.push(text.substring(partStart, index - 1));
+
+                    // Start after the asterisk and scan until a */ is encountered.
+                    index++;
+                    while (index < length) {
+                        if (peek() === CharCode.asterisk && peek(1) === CharCode.slash) {
+                            index += 2;
+                            break;
+                        }
+
+                        index++;
+                    }
+
+                    partStart = index;
+                }
+
+                break;
+        }
+
+        if (index >= length && index > partStart) {
+            parts.push(text.substring(partStart, length));
+            break;
+        }
+    }
+
+    return parts.join('');
+}
+
+export function tolerantParse(text: string) {
+    text = stripComments(text);
```

Comment:
OK. Added code to be tolerant against both byte order marks and trailing comments in object member or array element lists.